wsd/COOLWSD.cpp

```diff
@@ -1017,56 +1016,39 @@ void ForKitProcWSHandler::handleMessage(const std::vector<char> &data)
 {
     LOG_TRC("ForKitProcWSHandler: handling incoming [" << COOLProtocol::getAbbreviatedMessage(&data[0], data.size()) << "].");
 
-    std::stringstream ss(std::string(data.cbegin(), data.cend()));
-    std::string line;
-    StringVector tokens;
+    const std::string firstLine = COOLProtocol::getFirstLine(&data[0], data.size());
+    const StringVector tokens = StringVector::tokenize(firstLine.data(), firstLine.size());
 
-    auto parseCount = [this, &tokens](const std::string &countName)-> std::optional<uint>{
-
-        if (tokens.equals(0, countName))
-        {
-            try {
-                int count = std::stoi(tokens[1]);
-                if (count >= 0)
-                {
-                    return count;
-                }
-            } catch (std::invalid_argument const& ex) {
-            } catch (std::out_of_range const& ex) {
-            }
-            LOG_WRN("Invalid '" << countName << "' message received. Got:" << tokens[1]);
-            return 0;
-        }
+    if (tokens.equals(0, "faultcount"))
+    {
+        int segFaultcount;
+        int killedCount;
+        bool ok = true;
 
-        return std::nullopt;
-    };
+        ok &= tokens.getIntParam(1, segFaultcount);
+        ok &= tokens.getIntParam(2, killedCount);
 
-    while(std::getline(ss, line,'\n'))
-    {
-        tokens = StringVector::tokenize(line.data(), line.size());
+        if (ok) {
 
-        auto segFaultcountOpt = parseCount("segfaultcount");
-        if (segFaultcountOpt && segFaultcountOpt.value()) {
-            const auto segFaultcount = segFaultcountOpt.value();
-            Admin::instance().addSegFaultCount(segFaultcount);
-            LOG_INF(segFaultcount << " coolkit processes crashed with segmentation fault.");
-            SigUtil::addActivity("coolkit(s) crashed");
-            UnitWSD::get().kitSegfault(segFaultcount);
-        }
+            if (segFaultcount) {
+                Admin::instance().addSegFaultCount(segFaultcount);
```

Comment:
Can we not really just adapt the method to the Admin::instance to also take three parameters ?

Reply:
Which was this comment I guess =) lets just make a single message to the addActivity with all three counts etc.

Reply:
Done